### ISSUE TITLE
UnhandledPromiseRejectionWarning when downloading asset

### ISSUE THREAD

**Issue by Rochet2** [i3h/download-release-asset#7]
## Problems
1. When run, an UnhandledPromiseRejectionWarning error occurred.
2. The CI did not stop even after the error.

## Solutions
1. The error is 503, according to this https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/503
the service was likely just unavailable for a while. Retrying would probably solve the issue.
Consider using for example https://www.npmjs.com/package/axios-retry

2. The CI did not stop even if it had an error. It appears that on [this line](https://github.com/i3h/download-release-asset/blob/6a9870c8f1c561f9e67550d6177c31c2b1c49fef/src/download-release-asset.js#L86) the promise is not awaited.
Adding await should fix the issue, or alternatively one could bundle the promises from the loop using [Promise.all](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) and await that.

## Details
```
2022-10-05T19:37:03.0004767Z ##[group]Run i3h/download-release-asset@6a9870c8f1c561f9e67550d6177c31c2b1c49fef
2022-10-05T19:37:03.0005073Z with:
2022-10-05T19:37:03.0005270Z   owner: TrinityCore
2022-10-05T19:37:03.0005466Z   repo: TrinityCore
2022-10-05T19:37:03.0005671Z   tag: TDB335.22081
2022-10-05T19:37:03.0005882Z   file: /TDB_full.*\.7z/
2022-10-05T19:37:03.0006093Z   path: bin/check_install/bin
2022-10-05T19:37:03.0006566Z   token: ***
2022-10-05T19:37:03.0008041Z ##[endgroup]
2022-10-05T19:37:08.6434980Z (node:13612) UnhandledPromiseRejectionWarning: Error: Request failed with status code 503
2022-10-05T19:37:08.6436108Z     at createError (/home/runner/work/_actions/i3h/download-release-asset/6a9870c8f1c561f9e67550d6177c31c2b1c49fef/dist/index.js:1712:15)
2022-10-05T19:37:08.6436886Z     at settle (/home/runner/work/_actions/i3h/download-release-asset/6a9870c8f1c561f9e67550d6177c31c2b1c49fef/dist/index.js:10281:12)
2022-10-05T19:37:08.6437728Z     at RedirectableRequest.handleResponse (/home/runner/work/_actions/i3h/download-release-asset/6a9870c8f1c561f9e67550d6177c31c2b1c49fef/dist/index.js:26477:9)
2022-10-05T19:37:08.6438289Z     at RedirectableRequest.emit (events.js:314:20)
2022-10-05T19:37:08.6439220Z     at RedirectableRequest.module.exports.913.RedirectableRequest._processResponse (/home/runner/work/_actions/i3h/download-release-asset/6a9870c8f1c561f9e67550d6177c31c2b1c49fef/dist/index.js:27137:10)
2022-10-05T19:37:08.6440290Z     at ClientRequest.RedirectableRequest._onNativeResponse (/home/runner/work/_actions/i3h/download-release-asset/6a9870c8f1c561f9e67550d6177c31c2b1c49fef/dist/index.js:26814:10)
2022-10-05T19:37:08.6440814Z     at Object.onceWrapper (events.js:421:26)
2022-10-05T19:37:08.6441148Z     at ClientRequest.emit (events.js:314:20)
2022-10-05T19:37:08.6441562Z     at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:601:27)
2022-10-05T19:37:08.6442040Z     at HTTPParser.parserOnHeadersComplete (_http_common.js:122:17)
2022-10-05T19:37:08.6443478Z (node:13612) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
2022-10-05T19:37:08.6445364Z (node:13612) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by i3h** [comment]
Hi @Rochet2 , thanks for you report. I will have a look into this issue this week.

**Comment by Rochet2** [comment]
I made a PR with suggested fixes here: https://github.com/i3h/download-release-asset/pull/8

**Comment by i3h** [comment]
@Rochet2 

I have checked and merged PR #8.

Now, github actions will throw "Error: Request failed with status code 503" and stop immediately when the service is unavailable.

Thanks for your contributions.
